analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt

```diff
@@ -160,37 +136,82 @@ class MainActivity : AppCompatActivity() {
             param(FirebaseAnalytics.Param.ITEMS, arrayOf(itemJeggingsCart))
         }
         // [END log_purchase]
-
         // [START log_refund]
         analytics.logEvent(FirebaseAnalytics.Event.REFUND) {
             param(FirebaseAnalytics.Param.TRANSACTION_ID, "T12345")
             param(FirebaseAnalytics.Param.AFFILIATION, "Google Store")
             param(FirebaseAnalytics.Param.CURRENCY, "USD")
             param(FirebaseAnalytics.Param.VALUE, 9.99)
-
             // (Optional) for partial refunds, define the item ID and quantity of refunded items
             param(FirebaseAnalytics.Param.ITEM_ID, "SKU_123")
             param(FirebaseAnalytics.Param.QUANTITY, 1)
-
             param(FirebaseAnalytics.Param.ITEMS, arrayOf(itemJeggings))
         }
         // [END log_refund]
-
         // [START apply_promo]
-        val promoParams = Bundle().apply {
-            putString(FirebaseAnalytics.Param.PROMOTION_ID, "SUMMER_FUN")
-            putString(FirebaseAnalytics.Param.PROMOTION_NAME, "Summer Sale")
-            putString(FirebaseAnalytics.Param.CREATIVE_NAME, "summer2020_promo.jpg")
-            putString(FirebaseAnalytics.Param.CREATIVE_SLOT, "featured_app_1")
-            putString(FirebaseAnalytics.Param.LOCATION_ID, "HERO_BANNER")
-            putParcelableArray(FirebaseAnalytics.Param.ITEMS, arrayOf<Parcelable>(itemJeggings))
-        }
-
+        val promoParams =
+            Bundle().apply {
+                putString(FirebaseAnalytics.Param.PROMOTION_ID, "SUMMER_FUN")
+                putString(FirebaseAnalytics.Param.PROMOTION_NAME, "Summer Sale")
+                putString(FirebaseAnalytics.Param.CREATIVE_NAME, "summer2020_promo.jpg")
+                putString(FirebaseAnalytics.Param.CREATIVE_SLOT, "featured_app_1")
+                putString(FirebaseAnalytics.Param.LOCATION_ID, "HERO_BANNER")
+                putParcelableArray(FirebaseAnalytics.Param.ITEMS, arrayOf<Parcelable>(itemJeggings))
+            }
         // Promotion displayed
         analytics.logEvent(FirebaseAnalytics.Event.VIEW_PROMOTION, promoParams)
-
         // Promotion selected
         analytics.logEvent(FirebaseAnalytics.Event.SELECT_PROMOTION, promoParams)
         // [END apply_promo]
     }
+    // [START ad_impression_moPub]
+    override fun onImpression(adUnitId: String, impressionData: ImpressionData?) {
```

Comment:
Based on the [superclass](https://github.com/firebase/snippets-android/blob/master/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt#L12), `override` is not applicable (here and throughout).